### PR TITLE
[PDI-19808] - Connections keeps increasing when the transformation is…

### PR DESCRIPTION
**Pull request by renato-s**
… alternately executed on the Server and through REST API when connection pooling is enabled

@andreramos89 
@bcostahitachivantara 
@smmribeiro

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 47m 57s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2148**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5315/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: repository/src/main/java/org/pentaho/platform/repository/DatabaseHelper.java [hunk 1]
```diff
@@ -29,12 +29,12 @@
...
 import org.pentaho.platform.api.repository2.unified.data.node.DataProperty;
+import org.pentaho.platform.util.StringUtil;
 
 import java.io.Serializable;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 import java.util.Map.Entry;
-import java.util.Set;
+import java.util.stream.Collectors;
```
Comment (bcostahitachivantara) [inline repository/src/main/java/org/pentaho/platform/repository/DatabaseHelper.java:37]: Please validate and remove unsed imports

File: repository/src/main/java/org/pentaho/platform/repository/DatabaseHelper.java [hunk 2]
```diff
@@ -208,9 +208,13 @@ public IDatabaseConnection databaseMetaToDatabaseConnection( final DatabaseMeta
     databaseConnection.setQuoteAllFields( databaseMeta.isQuoteAllFields() );
     databaseConnection.setUsingDoubleDecimalAsSchemaTableSeparator( databaseMeta.isUsingDoubleDecimalAsSchemaTableSeparator() );
 
-    Map<String, String> attributeMap = new HashMap<>();
-    for ( Entry<Object, Object> entry: databaseMeta.getAttributes().entrySet() ) {
-      attributeMap.put( (String) entry.getKey(), databaseMeta.environmentSubstitute( (String) entry.getValue() ) );
+    Map<String, String> attributeMap = new TreeMap<>();
```
Comment (bcostahitachivantara) [inline repository/src/main/java/org/pentaho/platform/repository/DatabaseHelper.java:211]: Why did you revert the implementation using java stream API?

File: repository/src/main/java/org/pentaho/platform/repository/DatabaseHelper.java (hunk 1 above)
Comment (renato-s) [reply repository/src/main/java/org/pentaho/platform/repository/DatabaseHelper.java:37]: Imports optimized

**Comment by buildguy** [comment]
### :warning: Build finished in 1h 7m 38s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 2 test(s) failed:

<details><summary><b>org.pentaho.test.platform.repository.RepositoryTestCase.testNothing on testNothing(org.pentaho.test.platform.repository.RepositoryTestCase)</b> (click to expand)</summary><p>

```
HIBUTIL.ERROR_0004 - beginTransaction() failed.
```

</p></details>

<details><summary><b>org.pentaho.test.platform.repository.RepositoryTestCase.testNothing</b> (click to expand)</summary><p>

```
HIBUTIL.ERROR_0004 - beginTransaction() failed.
```

</p></details>

  
Tests run: **2874**, Failures: **2**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5315/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**